### PR TITLE
ci: updating sonarqube

### PR DESCRIPTION
**Pull request by FritzHoing**
## Description
SonarQube gets scanned for jest unit-tests and gets updated after every push on main branch. 

## Pull request type

- [ ] Bugfix
- [ ] Feature
- [ ] Dependency updates
- [ ] Tests
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Do you introduce a breaking change?

- [ ] Yes
- [ ] No

## Checklist

- [ ] I understand and agree that the changes in this PR will be licensed under the 
  [Apache Licence Version 2.0](https://github.com/terrestris/shogun/blob/main/LICENSE).
- [ ] I have followed the [guidelines for contributing](https://github.com/terrestris/shogun/blob/main/CONTRIBUTING.md).
- [ ] The proposed change fits to the content of the [code of conduct](https://github.com/terrestris/shogun/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have added or updated tests and documentation, and the test suite passes (run `mvn test` locally).
- [ ] I have added a screenshot/screencast to illustrate the visual output of my update.

### CODE REVIEW

**Review by LukasLohoff — COMMENTED** [review]
I've had two more ideas in the meantime:

- I think we could now use the same github action we used here: https://github.com/terrestris/shogun-admin/pull/103. This would make things more consistent
- please make use of `SONARQUBE_HOST` as well

File: .github/workflows/build-and-publish.yml [hunk 1]
```diff
@@ -42,3 +42,6 @@ jobs:
 
     - name: Build SHOGun with Maven and publish artifacts to nexus
       run: mvn -B deploy
+
+    - name: Refresh SonarQube
+      run: mvn compile sonar:sonar   -Dsonar.projectKey=SHOGun   -Dsonar.host.url=https://sq.terrestris.de   -Dsonar.login="${{ secrets.SONARQUBE_TOKEN }}"
```
Comment (dnlkoch) [inline .github/workflows/build-and-publish.yml:47]:
```suggestion
      run: mvn compile sonar:sonar -Dsonar.projectKey=SHOGun -Dsonar.host.url=https://sq.terrestris.de -Dsonar.login="${{ secrets.SONARQUBE_TOKEN }}"
```

File: .github/workflows/build-and-publish.yml [hunk 2]
```diff
@@ -42,3 +42,9 @@ jobs:
 
     - name: Build SHOGun with Maven and publish artifacts to nexus
       run: mvn -B deploy
+
+    - name: Scan for Unit Tests
+      run:  mvn clean verify sonar:sonar -Dsonar.projectKey=SHOGun -Dsonar.login="${{ secrets.SONARQUBE_TOKEN }}" -Preporting
```
Comment (LukasLohoff) [inline .github/workflows/build-and-publish.yml:47]:
```suggestion
      run: mvn clean verify sonar:sonar -Dsonar.projectKey=SHOGun -Dsonar.login="${{ secrets.SONARQUBE_TOKEN }}" -Preporting
```

File: pom.xml [hunk 3]
```diff
@@ -269,6 +279,26 @@
...
+          </execution>
+          <execution>
+            <id>report</id>
+            <phase>test</phase>
+            <goals>
+              <goal>report</goal>
+            </goals>
+          </execution>
+        </executions>
+          </plugin>
```
Comment (LukasLohoff) [inline pom.xml:301]:
```suggestion
      </plugin>
```

**Review by LukasLohoff — APPROVED** [review]

**Pull request closed without merging**